block-layer/codecs/dag-pb.md

```diff
@@ -0,0 +1,59 @@
+# DagPB Spec
+
+DagPB does not support the full ["IPLD Data Model."](../../data-model-layer/data-model.md)
+
+## Format
+
+The DagPB IPLD format is a legacy format implemented with a single protobuf.
+
+```protobuf
+// An IPFS MerkleDAG Link
+message PBLink {
+
+  // binary CID (with no multibase prefix) of the target object
+  optional bytes Hash = 1;
+
+  // UTF-8 string name.
+  optional string Name = 2;
+
+  // cumulative size of target object
+  optional uint64 Tsize = 3;
+}
+
+// An IPFS MerkleDAG Node
+message PBNode {
+
+  // refs to other objects
+  repeated PBLink Links = 2;
+
+  // opaque user data
+  optional bytes Data = 1;
+}
+```
+
+The objects link names are specified in the 'Name' field of the PBLink object.
+All link names in an object must either be blank or unique within the object.
+
+## Pathing
```

Comment:
I don't think discussing differences between implementations makes for a good spec.  If you were implementing this, what would you implement here?

Maybe just standardise on the JS version since it lets you navigate non-named links and the go version doesn't?

Reply:
As per https://github.com/ipld/specs/pull/201/files/d7f528343fa4b051a7ab1042013a4830a8f090a5#r330630459, I'll add a link to the bug. But I think it's better to know how things are currently implemented, rather than not having this documented at all.